cmake/compiler_flags_GNU_Fortran.cmake

```diff
@@ -8,25 +8,25 @@
 # FLAGS COMMON TO ALL BUILD TYPES
 ####################################################################
 
-set( CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -g -ffree-line-length-none")
+set( CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -g -ffree-line-length-none -mcmodel=medium")
```

Comment:
think the `mcmodel` is a Intel compiler flag not `gnu` I do not see it in the gnu flags

Reply:
@BenjaminRuston it's a GNU flag. Both intel and gnu used the same backend up until recently. Intel is moving to clang (may have already done so with OneAPI). It's a cpu dependent  flag for arm and x86.
https://gcc.gnu.org/onlinedocs/gcc/Option-Summary.html

Reply:
@PatNichols You probably saw that pushed those flags for Intel only directly in the CMakeLists in the obs2ioda subdirectory. Feel free to take those out and make them consistent with the gnu flags, or vice versa. Thanks!

Reply:
@climbfuji Thanks! I have migrated those changes into our cmake/compiler_flags.